python/ray/dag/dag_node_operation.py

```diff
@@ -297,17 +375,98 @@ def _build_dag_node_operation_graph(
             _add_edge(
                 graph[task_idx][_DAGNodeOperationType.WRITE],
                 graph[downstream_task_idx][_DAGNodeOperationType.READ],
+                "nccl"
+                if graph[task_idx][_DAGNodeOperationType.WRITE].requires_nccl
+                else "shm",
```

Comment:
nit: If the reader and writer belong to the same actor, "shm" might be incorrect.

Reply:
yeah I think this is fine for now. We can refine this later.